data-runtime/src/main/java/io/micronaut/data/runtime/intercept/AbstractQueryInterceptor.java

```diff
@@ -507,11 +508,15 @@ protected Object instantiateEntity(@NonNull Class<?> rootEntity, @NonNull Map<St
                     PersistentProperty prop = entity.getPropertyByName(argumentName);
                     if (prop == null || prop.isRequired()) {
                         throw new IllegalArgumentException("Argument [" + argumentName + "] cannot be null");
+                    } else {
+                        // If Optional or AutoGenerated or AutoPopulated with null value
+                        // is one of fields, then we shouldn't force strict nullability check
+                        strictNullable = false;
                     }
                 }
                 arguments[i] = v;
             }
-            instance = introspection.instantiate(arguments);
+            instance = introspection.instantiate(strictNullable, arguments);
```

Comment:
This is one way to avoid the exception when there is `@AutoPopulated` field, don't do strict null check. Shouldn't be an issue because there is the check present here for required non nullable field with null value